src/helper.cc

```diff
@@ -557,19 +557,20 @@ helper::prepSubmit()
     return false;
 }
 
+/// If possible, submit request. Otherwise, either kill Squid or return false.
 bool
-helper::trySubmit(const char *buf, HLPCB * callback, void *data)
+helper::trySubmit(const char *buf, HLPCB * callback, void *data, const Helper::ReservationId &reservation)
 {
     if (!prepSubmit())
         return false; // request was dropped
 
-    submit(buf, callback, data); // will send or queue
+    submit(buf, callback, data, reservation); // will send or queue
     return true; // request submitted or queued
 }
 
 /// dispatches or enqueues a helper requests; does not enforce queue limits
 void
-helper::submit(const char *buf, HLPCB * callback, void *data)
+helper::submit(const char *buf, HLPCB * callback, void *data, const Helper::ReservationId &)
```

Comment:
None of the concepts represented by the current `::helper` class should know about helper reservations. Reservation is a concept exclusive to the `statefulhelper` class. We should not add reservation-specific parameters to `::helper`, even if these reservations are going to be always nil for stateless helpers.

Reply:
Well then. Since the objected-to detail is a hard prerequisite of this PR change I am closing it entirely. Will see what can be done after the @eduard-bagdasaryan  changes.